benchmarks/babelstream/src/babelStreamMainTest.cpp

```diff
@@ -267,210 +296,328 @@ void testKernels()
+            if constexpr(alpaka::
+                             accMatchesTags<TAcc, alpaka::TagGpuCudaRt, alpaka::TagGpuHipRt, alpaka::TagGpuSyclIntel>)
+            {
+                using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+                // Threads per block for Dot kernel
+                constexpr Idx blockThreadExtent = blockThreadExtentMain;
+                // Blocks per grid for Dot kernel
+                Idx const gridBlockExtent = static_cast<Idx>(dotGridBlockExtent);
+                // Vector of sums of each block
+                auto bufAccSumPerBlock = alpaka::allocBuf<DataType, Idx>(devAcc, gridBlockExtent);
+                auto bufHostSumPerBlock = alpaka::allocBuf<DataType, Idx>(devHost, gridBlockExtent);
+                // A specific work-division is used for dotKernel
+                auto const workDivDot = WorkDiv{Vec{gridBlockExtent}, Vec{blockThreadExtent}, Vec::all(1)};
+
+                measureKernelExec(
+                    [&]()
+                    {
+                        alpaka::exec<Acc>(
+                            queue,
+                            workDivDot,
+                            DotKernel(), // Dot kernel
+                            bufAccInputAPtr,
+                            bufAccInputBPtr,
+                            alpaka::getPtrNative(bufAccSumPerBlock),
+                            static_cast<alpaka::Idx<Acc>>(arraySize));
+                        alpaka::memcpy(queue, bufHostSumPerBlock, bufAccSumPerBlock, gridBlockExtent);
+                        alpaka::wait(queue);
+                        DataType const* sumPtr = std::data(bufHostSumPerBlock);
+                        resultDot = std::reduce(sumPtr, sumPtr + gridBlockExtent, 0.0f);
+                    },
+                    "DotKernel");
+                // Add workdiv to the list of workdivs to print later
+                metaData.setItem(BMInfoDataType::WorkDivDot, workDivDot);
+            }
+        }
+        else if(kernelsToBeExecuted == KernelsToRun::NStream)
+        {
+            // Test the NStream-kernel. Calculate A += B + scalar * C;
+            measureKernelExec(
+                [&]() {
+                    alpaka::exec<Acc>(
+                        queue,
+                        workDivNStream,
+                        NstreamKernel(),
+                        bufAccInputAPtr,
+                        bufAccInputBPtr,
+                        bufAccOutputCPtr);
+                },
+                "NStreamKernel");
+        }
+        alpaka::wait(queue);
+
+    } // End of main loop which runs the kernels many times
+
+
+    // Copy arrays back to host since the execution of kernels except dot kernel finished
     alpaka::memcpy(queue, bufHostOutputC, bufAccOutputC, arraySize);
     alpaka::memcpy(queue, bufHostOutputB, bufAccInputB, arraySize);
     alpaka::memcpy(queue, bufHostOutputA, bufAccInputA, arraySize);
```

Comment:
These get timed [int the original version](https://github.com/UoB-HPC/BabelStream/blob/2f00dfb7f8b7cfe8c53d20d5c770bccbf8673440/src/main.cpp#L338).

Reply:
Read time (actually copy time from Acc to Host for 3 arrays ) has been added to the output display as **AccToHost Memcpy Time(sec).** Thanks.

```
AcceleratorType:AccGpuCudaRt<1,unsigned int>
NumberOfRuns:100
Precision:double
DataSize(items):33554432
DeviceName:NVIDIA RTX A500 Laptop GPU
WorkDivInit :{gridBlockExtent: (32768), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivCopy :{gridBlockExtent: (32768), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivMult :{gridBlockExtent: (32768), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivAdd  :{gridBlockExtent: (32768), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivTriad:{gridBlockExtent: (32768), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivDot  :{gridBlockExtent: (256), blockThreadExtent: (1024), threadElemExtent: (1)}
AccToHost Memcpy Time(sec):0.570856
Kernels         Bandwidths(GB/s) MinTime(s) MaxTime(s) AvgTime(s) DataUsage(MB) 
 AddKernel       90.665          0.0088822 0.0089985 0.0089376 805.31 
 CopyKernel      89.087          0.0060264 0.0061119 0.0060773 536.87 
 DotKernel       93.055          0.0057694 0.0058486 0.0058113 536.87 
 InitKernel      84.437          0.0095374 0.0095374 0.0095374 805.31 
 MultKernel      89.35           0.0060086 0.0060852 0.0060568 536.87 
 TriadKernel     90.222          0.0089258 0.0090338 0.0089565 805.31 

```